pkg/controller/hyperconverged/upgradePatches.go

```diff
@@ -0,0 +1,92 @@
+package hyperconverged
+
+import (
+	"encoding/json"
+	"errors"
+	"github.com/blang/semver/v4"
+	jsonpatch "github.com/evanphx/json-patch"
+	"github.com/kubevirt/hyperconverged-cluster-operator/pkg/controller/common"
+	hcoutil "github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
+	"os"
+	"strings"
+)
+
+const (
+	upgradeChangesFileLocation = "./upgradePatches.yaml"
+)
+
+type hcoCRPatch struct {
+	// SemverRange is a set of conditions which specify which versions satisfy the range
+	// (see https://github.com/blang/semver#ranges as a reference).
+	SemverRange string `json:"semverRange"`
+	// JSONPatch contains a sequence of operations to apply to the HCO CR during upgrades
+	// (see: https://datatracker.ietf.org/doc/html/rfc6902 as the format reference).
+	JSONPatch string `json:"jsonPatch"`
+}
+
+type upgradePatches struct {
+	// hcoCRPatchList is a list of upgrade patches.
+	// Each hcoCRPatch consists in a semver range of affected source versions and a json patch to be applied during the upgrade if relevant.
+	HCOCRPatchList []hcoCRPatch `json:"hcoCRPatchList"`
+}
+
```

Comment:
Can also be:

```suggestion
// upgradePatches is a list of upgrade patches.
// Each hcoCRPatch consists in a semver range of affected source versions and a json patch to be applied type 
upgradePatches struct HCOCRPatchList []hcoCRPatch
```

Reply:
I prefer to keep it in this way to keep places to add other operators like deleting other objects in the future